### PR TITLE
feat: Add more information to touch breadcrumbs

### LINKED ISSUE #1723: Include title in some events, like touch?
### Problem Statement

Right now when e.g. a button is tapped, it takes some sleuthing to figure out which button was tapped. Is it possible to include e.g. a view's title or tag in order to make identification easier?

### Solution Brainstorm

Right now we get this in the breadcrumb:

```
{
view: <UIButton: 0x160be6d90; frame = (0 0; 368 44); opaque = NO; tintColor = UIExtendedSRGBColorSpace 0.447059 0.670588 0.74902 1; layer = <CALayer: 0x2824adec0>>
}
```

It would be helpful if e.g. `title` and/or `tag` could be included, if not nil or 0, like this:

```
{
view: <UIButton: 0x160be6d90; frame = (0 0; 368 44); opaque = NO; tintColor = UIExtendedSRGBColorSpace 0.447059 0.670588 0.74902 1; layer = <CALayer: 0x2824adec0>>,
title: "Start",
tag: 1234
}
```

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Added more context in the `touch` breadcrumb. Now we have the `title`, `tag`, and `accessibilityIdentifier` of the view that was tapped.

image

## :bulb: Motivation and Context

close #1723 

## :green_heart: How did you test it?

Samples

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+- feat: Add more information on the action breadcrumb (#1724)
```
Comment (philipphofmann) [inline CHANGELOG.md:5]: `l`: 
```suggestion
- feat: Add more info to touch event breadcrumbs (#1724)
```

File: Sources/Sentry/SentryBreadcrumbTracker.m [hunk 2]
```diff
@@ -134,12 +134,35 @@ - (void)swizzleSendAction
...
-                        data = @ { @"view" : [NSString stringWithFormat:@"%@", touch.view] };
+                        data = @ { @"view" : [NSString stringWithFormat:@"%@", touch.view] }
+                                   .mutableCopy;
+
+                        if ([sender isKindOfClass:UIView.class]) {
+                            UIView *senderView = sender;
+                            if (senderView.tag > 0) {
+                                [data setValue:[NSNumber numberWithInteger:senderView.tag]
+                                        forKey:@"tag"];
+                            }
```
Comment (philipphofmann) [inline Sources/Sentry/SentryBreadcrumbTracker.m:148]: `m`: I wonder if adding the tag actually makes sense. Would it help you when you debug your app?

File: Sources/Sentry/SentryBreadcrumbTracker.m [hunk 3]
```diff
@@ -134,12 +134,35 @@ - (void)swizzleSendAction
...
             if (nil != [SentrySDK.currentHub getClient]) {
-                NSDictionary *data = nil;
+                NSMutableDictionary *data = nil;
                 for (UITouch *touch in event.allTouches) {
                     if (touch.phase == UITouchPhaseCancelled || touch.phase == UITouchPhaseEnded) {
-                        data = @ { @"view" : [NSString stringWithFormat:@"%@", touch.view] };
+                        data = @ { @"view" : [NSString stringWithFormat:@"%@", touch.view] }
+                                   .mutableCopy;
+
+                        if ([sender isKindOfClass:UIView.class]) {
```
Comment (philipphofmann) [inline Sources/Sentry/SentryBreadcrumbTracker.m:143]: `m`: Can we add some tests for this new logic, please? The easiest way to make this testable would be to add an extra function for this logic and just call it directly from the tests and validate the data, for example.

**Review by philipphofmann — COMMENTED** [review]
Thanks, for doing this 👏

File: Sources/Sentry/SentryBreadcrumbTracker.m (hunk 2 above)
Comment (brustolin) [reply Sources/Sentry/SentryBreadcrumbTracker.m:148]: I can thing on some very specifics cases, like dynamic generated fixed UI.

**Review by philipphofmann — APPROVED** [review]
Thanks for the tests 👏

**Comment by matthewreimer** [comment]
This is great, thanks! 👏

**Pull request merged**